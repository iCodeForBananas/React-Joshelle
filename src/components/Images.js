// @flow
import React from "react";
import "./Images.css";
import Image from "./Image";
import sampleSize from "lodash/sampleSize";
import range from "lodash/range";

type Props = {};

type State = {
  currentImages: {
    left: number,
    top: number,
    src: number
  }[]
};

class Images extends React.Component<Props, State> {
  images: Array<number> = range(1, 60);

  state = {
    currentImages: []
  };

  componentDidMount() {
    this.updateImages();
  }

  randomIntFromInterval(min: number = 0, max: number = 100): number {
    return Math.floor(Math.random() * (max - min + 1) + min);
  }

  updateImages() {
    const images = sampleSize(this.images, 10);

    this.setState(
      {
        currentImages: images.map((image: number) => ({
          top: Math.max(0, this.randomIntFromInterval(0, window.innerHeight) - 200),
          left: Math.max(0, this.randomIntFromInterval(0, window.innerWidth) - 200),
          src: image
        }))
      },
      () => {
        setTimeout(() => this.updateImages(), 3450);
      }
    );
  }

  render() {
    return (
      <React.Fragment>
        {this.state.currentImages.map(image => {
          const src = "/images/" + image.src + ".gif";
          return <Image top={image.top} left={image.left} key={src} src={src} />;
        })}
      </React.Fragment>
    );
  }
}

export default Images;
