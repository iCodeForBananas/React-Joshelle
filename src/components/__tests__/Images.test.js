import React from "react";
import Images from "../Images";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Images />).toJSON();
  expect(tree).toMatchSnapshot();
});
