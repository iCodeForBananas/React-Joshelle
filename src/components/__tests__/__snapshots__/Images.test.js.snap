// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/54.gif"
    style={
        Object {
            "marginLeft": "307px",
            "marginTop": "6px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/5.gif"
    style={
        Object {
            "marginLeft": "722px",
            "marginTop": "567px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/34.gif"
    style={
        Object {
            "marginLeft": "304px",
            "marginTop": "326px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/28.gif"
    style={
        Object {
            "marginLeft": "227px",
            "marginTop": "477px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/11.gif"
    style={
        Object {
            "marginLeft": "316px",
            "marginTop": "0px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/33.gif"
    style={
        Object {
            "marginLeft": "544px",
            "marginTop": "425px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/39.gif"
    style={
        Object {
            "marginLeft": "536px",
            "marginTop": "306px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/49.gif"
    style={
        Object {
            "marginLeft": "287px",
            "marginTop": "284px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/13.gif"
    style={
        Object {
            "marginLeft": "17px",
            "marginTop": "171px",
          }
    }
/>,
  <img
    alt=""
    className="Image-img Image-spin"
    src="/images/42.gif"
    style={
        Object {
            "marginLeft": "687px",
            "marginTop": "0px",
          }
    }
/>,
]
`;
